vitest.config.ts

```diff
@@ -0,0 +1,39 @@
+/// <reference types="vitest" />
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+import path from 'path'
+
+export default defineConfig({
+  plugins: [react()],
+  test: {
+    globals: true,
+    environment: 'happy-dom',
+    // setupFiles: ['./src/test/setup.ts'],
+    css: true,
+    // Include coverage configuration
+    coverage: {
+      provider: 'v8',
+      reporter: ['text', 'json', 'html'],
+      exclude: [
+        'node_modules/',
+        'src/test/',
+        '**/*.d.ts',
+        '**/*.config.*',
+        'src/vite-env.d.ts'
+      ]
+    },
+    // Mock WebAssembly for tests
+    pool: 'forks',
+    poolOptions: {
+      forks: {
+        singleFork: true
+      }
+    }
+  },
+  resolve: {
+    alias: {
+      '@': path.resolve(__dirname, './src'),
+      "monaco-editor": "./src/__mocks__/monaco-editor.ts"
```

Comment:
the docs recommend using `test.alias` rather than `resolve.alias` for mocks. I'm not sure that there's any mechanical different, but the separation of concerns does seem kind of nice.

https://vitest.dev/guide/mocking.html#modules